print-babel.c

```diff
@@ -556,17 +561,19 @@ babel_print_v2(netdissect_options *ndo,
             else {
                 int rc;
                 u_short seqno;
-                u_char prefix[16], plen;
+                u_char prefix[16], ae, plen;
                 ND_PRINT("\n\tMH-Request ");
                 if(len < 14) goto invalid;
+                ae = EXTRACT_U_1(message + 2);
                 seqno = EXTRACT_BE_U_2(message + 4);
-                rc = network_prefix(EXTRACT_U_1(message + 2),
+                rc = network_prefix(ae,
                                     EXTRACT_U_1(message + 3), 0,
                                     message + 16, NULL, len - 14, prefix);
                 if(rc < 0) goto invalid;
                 plen = EXTRACT_U_1(message + 3) + (EXTRACT_U_1(message + 2) == 1 ? 96 : 0);
                 ND_PRINT("(%u hops) for %s seqno %u id %s",
-                       EXTRACT_U_1(message + 6), format_prefix(ndo, prefix, plen),
+                       EXTRACT_U_1(message + 6),
+                       ae == 0 ? "invalid" : format_prefix(ndo, prefix, plen),
```

Comment:
Would it be more helpful to print something like "invalid AE 0"? Also `MESSAGE_NH` likely needs something similar. I understand the fixup is not required for `MESSAGE_UPDATE_SRC_SPECIFIC`, `MESSAGE_REQUEST_SRC_SPECIFIC` and `MESSAGE_MH_REQUEST_SRC_SPECIFIC` because those TLV disappear after draft-boutier-babel-source-specific-01 (May 2015) and the whole case blocks are now obsolete.